### PR TITLE
add mesh class

### PR DESCRIPTION
**Pull request by mleotta**
Mesh classes and file IO ported from VXL and adapted to KWIVER dependencies and style.  Tests are still needed.

### CODE REVIEW

File: vital/io/mesh_io.cxx [hunk 1]
```diff
@@ -0,0 +1,819 @@
...
+
+  //write texture coordinates
+  if (mesh.has_tex_coords() == mesh::TEX_COORD_ON_VERT) {
+    os << " texCoord TextureCoordinate {\n"
+       << "   point [\n";
+
+    //write tex coordinates (should be same number as vertices above)
+    const std::vector<vector_2d >& tc = mesh.tex_coords();
+    for (unsigned int i=0; i<tc.size(); ++i)
+      os << "    " << tc[i].x() << ' ' << tc[i].y() << ",\n";
```
Comment (linus-sherrill) [inline vital/io/mesh_io.cxx]: Braces around single line block.

File: vital/types/mesh.cxx [hunk 2]
```diff
@@ -0,0 +1,628 @@
...
+  }
+  else if (this->has_tex_coords() == other.has_tex_coords())
+  {
+    this->tex_coords_.insert(this->tex_coords_.end(),
+                             other.tex_coords().begin(),
+                             other.tex_coords().end());
+  }
+
+  if (this->has_half_edges() && other.has_half_edges())
+    this->build_edge_graph();
```
Comment (linus-sherrill) [inline vital/types/mesh.cxx]: Braces around single line block.

File: vital/types/mesh.cxx [hunk 3]
```diff
@@ -0,0 +1,628 @@
...
+  else
+    this->half_edges_.clear();
```
Comment (linus-sherrill) [inline vital/types/mesh.cxx]: Braces around single line block.

File: vital/types/mesh.cxx [hunk 4]
```diff
@@ -0,0 +1,628 @@
...
+}
+
+
+/// Set the texture coordinates
+void
+mesh
+::set_tex_coords(const std::vector<vector_2d>& tc)
+{
+  if (tc.size() == this->num_verts())
+    tex_coord_status_ = TEX_COORD_ON_VERT;
```
Comment (linus-sherrill) [inline vital/types/mesh.cxx]: Braces around single line block, and below.

File: vital/types/mesh.cxx [hunk 5]
```diff
@@ -0,0 +1,628 @@
...
+void
+mesh
+::build_edge_graph()
+{
+  const mesh_face_array_base& faces = this->faces();
+  std::vector<std::vector<unsigned int> > face_list(faces.size());
+  for (unsigned int f=0; f<faces.size(); ++f) {
+    face_list[f].resize(faces.num_verts(f));
+    for (unsigned int v=0; v<faces.num_verts(f); ++v)
+      face_list[f][v] =  faces(f,v);
```
Comment (linus-sherrill) [inline vital/types/mesh.cxx]: Braces around single line block.

File: vital/types/mesh.cxx [hunk 6]
```diff
@@ -0,0 +1,628 @@
...
+}
+
+
+/// Compute vertex normals
+void
+mesh
+::compute_vertex_normals()
+{
+  if (!this->has_half_edges())
+    this->build_edge_graph();
```
Comment (linus-sherrill) [inline vital/types/mesh.cxx]: Braces around single line block.

File: vital/types/mesh.cxx [hunk 7]
```diff
@@ -0,0 +1,628 @@
...
+
+  const mesh_half_edge_set& half_edges = this->half_edges();
+  mesh_vertex_array<3>& verts = this->vertices<3>();
+
+  std::vector<vector_3d> normals(this->num_verts(), vector_3d(0,0,0));
+
+  for (unsigned int he=0; he < half_edges.size(); ++he) {
+    mesh_half_edge_set::f_const_iterator fi(he,half_edges);
+    if (fi->is_boundary())
+      continue;
```
Comment (linus-sherrill) [inline vital/types/mesh.cxx]: Braces around single line block.

File: vital/types/mesh.cxx [hunk 8]
```diff
@@ -0,0 +1,628 @@
...
+}
+
+
+/// Compute vertex normals using face normals
+void
+mesh
+::compute_vertex_normals_from_faces()
+{
+  if (!this->has_half_edges())
+    this->build_edge_graph();
```
Comment (linus-sherrill) [inline vital/types/mesh.cxx]: Braces around single line block.
I'll stop commenting on this at this point, but there are still some single statement blocks that need braces further down.

File: vital/types/mesh.cxx [hunk 9]
```diff
@@ -0,0 +1,628 @@
...
+vector_2d
+mesh
+::texture_map(unsigned int tri, double u, double v) const
+{
+  vector_2d tex(0,0);
+  if (this->tex_coord_status_ == TEX_COORD_ON_VERT)
+  {
+    unsigned int v1 = (*faces_)(tri,0);
+    unsigned int v2 = (*faces_)(tri,1);
+    unsigned int v3 = (*faces_)(tri,2);
```
Comment (linus-sherrill) [inline vital/types/mesh.cxx]: Cold these be const?

**Comment by mleotta** [comment]
Linus, thanks for the review.  FYI, this is code ported from VXL, and mostly written by me while in grad school.  That explains some of the style differences like braces.  I tried to adapt to KWIVER style, but clearly missed some cases.  I will go back and correct those.  I'll also convert cerr/cout to logger messages, I forgot to do that.   

In short, I'll run through your comments and address them in new commits.

**Comment by mleotta** [comment]
@linus-sherrill, I think I've addressed all of the items from your review

**Pull request merged**